superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover.jsx

```diff
@@ -125,7 +125,15 @@ export default class AdhocMetricEditPopover extends React.Component {
       label = metricLabel;
     }
 
-    const metric = savedMetric?.metric_name ? savedMetric : adhocMetric;
+    // if saved metric, generate a unique savedMetricId
+    const metric = savedMetric?.metric_name
+      ? {
+          ...savedMetric,
+          savedMetricId: `${Math.random()
```

Comment:
This unique ID seems somewhat hard to grok. Why not use a UUID?

Reply:
I did not want to add another dependency to the project and that method of generating ID was already in use in `AdhocMetric.js`